qiskit/ignis/verification/accreditation/qotp.py

```diff
@@ -0,0 +1,265 @@
+    dag_internal = circuit_to_dag(circ_internal)
+    for dag_layer in dag_internal.layers():
+        circuit_layer = dag_to_circuit(dag_layer['graph'])
+        for circelem, qsub, csub in circuit_layer:
+            n = circelem.name
+            if n == "barrier":
+                # if a barrier separates any two qubit gates
+                # start a new layer
+                if current2qs != []:
+                    singlequbitlayers.append(QuantumCircuit(qregs, cregs))
+                    twoqubitlayers.append(QuantumCircuit(qregs, cregs))
+                    current2qs = []
+                singlequbitlayers[-2].append(circelem, qsub, csub)
+            elif n in ('u1', 'u2', 'u3'):
+                # single qubit gate
+                q = qsub[0]
+                if q in current2qs:
+                    singlequbitlayers[-1].append(circelem, qsub, csub)
+                else:
+                    singlequbitlayers[-2].append(circelem, qsub, csub)
+            elif n == "cx":
+                # cx indices
+                q0 = qsub[0]
+                q1 = qsub[1]
+                # check if new cnot satisfies overlap criteria
+                if q0 in current2qs or q1 in current2qs:
+                    singlequbitlayers.append(QuantumCircuit(qregs, cregs))
+                    twoqubitlayers.append(QuantumCircuit(qregs, cregs))
+                    current2qs = []
+                if two_qubit_gate == 'cx':
+                    # append cx
+                    twoqubitlayers[-1].cx(q0, q1)
+                elif two_qubit_gate == 'cz':
+                    # append and correct to cz with h gates
+                    twoqubitlayers[-1].cz(q0, q1)
+                    singlequbitlayers[-1].h(qsub[1])
+                    singlequbitlayers[-2].h(qsub[1])
+                else:
+                    raise Exception("Two qubit gate {0}".format(two_qubit_gate)
+                                    + " is not implemented in qotp")
+                # add to current
+                current2qs.append(q0)
+                current2qs.append(q1)
+            elif n == "measure":
+                measlayer.append(circelem, qsub, csub)
+            else:
+                raise Exception("Circuit element {0}".format(n)
+                                + " is not implemented in qotp")
+    if current2qs == []:
+        del singlequbitlayers[-1]
+        del twoqubitlayers[-1]
+    for ind, circlayer in enumerate(singlequbitlayers):
+        singlequbitlayers[ind] = transpile(circlayer,
+                                           basis_gates=['u1', 'u2', 'u3'])
+    parsedlayers = {'singlequbitlayers': singlequbitlayers,
+                    'twoqubitlayers': twoqubitlayers,
+                    'measlayer': measlayer,
+                    'twoqubitgate': two_qubit_gate,
+                    'qregs': qregs,
+                    'cregs': cregs}
+    return parsedlayers
+
+
+def QOTP_fromlayers(layers):
```

Comment:
So you can pass the RNG in from the AccreditationCircuit class:

```suggestion
def QOTP_fromlayers(layers, rng=None):
```

Then in the class you could have the case for when this is None:
```python
if rng is None:
    rng = random.Random()  # or np.random.RandomState()
```